src/datachain/query/dispatch.py

```diff
@@ -224,6 +219,9 @@ def run_udf_parallel(  # noqa: C901, PLR0912
             input_finished = False
 
             if not streaming_mode:
+                if not self.is_batching:
+                    input_rows = batched(flatten(input_rows), DEFAULT_BATCH_SIZE)
```

Comment:
Batching and flatten IDs for better performance so each parallel process will receive batch of IDs and can select them all at once instead of many single selects.

Here is it better to analyze input dataset size (`count`) and adjust batch size because on small input datasets it can be one parallel process who get all the rows IDs. Item for the follow-up issue, I guess.